spec/connectors/registry_spec.rb

```diff
@@ -9,21 +9,74 @@
 require 'connectors/registry'
 
 describe Connectors::Factory do
-  let(:factory) do
-    described_class.new
+  subject { described_class.new }
+
+  let(:configuration) {
+    {}
+  }
+
+  let(:job_description) {
+    {}
+  }
+  class MyConnector
+    def initialize(configuration: {}, job_description: {})
+      # nothing needed
+    end
+  end
+
+  before(:each) do
+    subject.register('my-connector', MyConnector)
   end
 
   describe '#connector_class' do
     it 'let us register a connector under a name' do
-      class MyConnector
-        def works
-          'works'
-        end
+      connector_class = subject.connector_class('my-connector')
+      expect(connector_class == (MyConnector)).to eq(true)
+    end
+  end
+
+  describe '#registered?' do
+    context 'my-connector is registered' do
+      it 'should return that sharepoint connector is registered' do
+        expect(subject.registered?('my-connector')).to eq(true)
+      end
+
+      it 'should return that another-connector is not registered' do
+        expect(subject.registered?('another-connector')).to eq(false)
       end
+    end
+  end
+
+  describe '#connector' do
+    context 'my-connector is registered' do
+      it 'should return a my-connector instance' do
+        connector_instance = subject.connector('my-connector', configuration, job_description)
+
+        expect(connector_instance).to_not be_nil
+        expect(connector_instance.is_a?(MyConnector)).to eq(true)
+      end
+    end
+
+    context 'another-connector is not registered' do
+      it 'should raise an exception, that another-connector is not registered' do
+        expect { subject.connector('sharepoint', configuration, job_description) }.to raise_exception
+      end
+    end
+  end
+
+  describe '#registered_connectors' do
+    before(:each) do
+      subject.register('b-connector', MyConnector)
+      subject.register('c-connector', MyConnector)
+      subject.register('a-connector', MyConnector)
+    end
+
+    it 'returns registered connectors sorted' do
+      registered_connectors = subject.registered_connectors
 
-      factory.register('sharepoint', MyConnector)
-      connector_class = factory.connector_class('sharepoint')
-      expect(connector_class.new.works).to eq 'works'
+      expect(registered_connectors[0]).to eq('a-connector')
+      expect(registered_connectors[1]).to eq('b-connector')
+      expect(registered_connectors[2]).to eq('c-connector')
```

Comment:
Might be better to just do `expect(registered_connectors).to include('a-connector')` to not make the test flakey if ordering of registration changes.

I'd probably also move `a-connector`, `b-connector` and `c-connector` to variables in test setup.

Reply:
👍